### ISSUE TITLE
User - Audit

### ISSUE THREAD

**Issue by mike-webster** [mike-webster/pet-tracker#36]
## Description
Go through the user functionality and take down what works and what doesn't.

## Technical Details
- Check for bugs
- Compile list of features to implement
    - Write up cards

**Comment by mike-webster** [comment]
No test coverage - at this point, we don't have much to unit test.

**Comment by mike-webster** [comment]
Settings page in the nav goes to `new_pet_path` because the settings page hasn't actually been implemented yet.

Settings page requirements:

Updates: 
- first name
- last name
- email
- password
- timezone

Delete account functionality - erase all user data.

**Comment by mike-webster** [comment]
Forgot password flow

**Comment by mike-webster** [comment]
Implement sharing with social media? Super low priority - let's try to make something people want to share before building the ability to do it.

**Comment by mike-webster** [comment]
Implement user tracking: 
Add tracking table to the database to track where users came to from.

**Comment by mike-webster** [comment]
Bad sign-in blocker - only allow a user x attempts to sign in before their account gets locked.

In order to get unlocked, they just need to interact with an email.

Requirements:
- User from the same IP address submits invalid credentials for the same user X times within 15 minutes, the account gets blocked from signing in for 1 hour.
-  When the account gets blocked, an email is sent to the user with a link in case they need to access their account sooner.
- Set up a route to use with the email
    - When the route is requested, we unblock the user's account and send them to the login page.
